sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/QueryPlanner.java

```diff
@@ -26,8 +26,29 @@
  */
 public interface QueryPlanner {
   /** It parses and validate the input query, then convert into a {@link BeamRelNode} tree. */
-  BeamRelNode convertToBeamRel(String sqlStatement) throws ParseException, SqlConversionException;
+  BeamRelNode convertToBeamRel(
+      String sqlStatement, Object queryParameters, QueryParameterMode queryParameterMode)
+      throws ParseException, SqlConversionException;
 
   /** Parse input SQL query, and return a {@link SqlNode} as grammar tree. */
   SqlNode parse(String sqlStatement) throws ParseException;
+
+  /**
+   * Parameters have different usage semantics depending on the parameter mode specified. Note that
+   * some parameter modes might not be supported by all backends.
+   */
+  enum QueryParameterMode {
```

Comment:
Is a query allows having both named and positional parameters?

Reply:
No. I updated the javadoc to clarify this.

Reply:
No. I updated the javadoc to clarify this.